### ISSUE TITLE
Latex Makefile not functional

### ISSUE THREAD

**Issue by bollenn** [doxygen/doxygen#9000]
Latex Makefile is not working since https://github.com/doxygen/doxygen/pull/8833

Because of the referenced MR I now get:

```sh
'make' --directory=latex all
rm -f *.ps *.dvi *.aux *.toc *.idx *.ind *.ilg *.log *.out *.brf *.blg *.bbl refman.pdf
"lualatex -interaction=batchmode -halt-on-error -shell-escape" refman
/bin/sh: 1: lualatex -interaction=batchmode -halt-on-error -shell-escape: not found
```

Since I'm passing in my Doxyfile

```
LATEX_CMD_NAME = lualatex -interaction=batchmode -halt-on-error -shell-escape

or

LATEX_CMD_NAME = "lualatex -interaction=batchmode -halt-on-error -shell-escape"

or

LATEX_CMD_NAME = "TEXINPUTS=$(abspath .)/latex-templates//: lualatex -interaction=batchmode -halt-on-error -shell-escape"
```

And doxygen now generates a Makefile as in:

```mk
LATEX_CMD="lualatex -interaction=batchmode -halt-on-error -shell-escape"

...

refman.pdf: clean refman.tex
	$(LATEX_CMD) refman
```

**Comment by bollenn** [comment]
@albert-github probably this is related to the fact that I'm using `LATEX_CMD_NAME` variable to pass extra variables and arguments to the tool.

Is there a better/preferred way of doing that?

**Comment by albert-github** [comment]
I can reproduce the behavior, I indeed think that the problem is related to the extra variables / arguments. The double quotes are necessary for the cases that there is e.g. a space in the LATEX_CMD_NAME.

I think that the following can work (at least at my place it works) :
```
LATEX_CMD_NAME = "\"lualatex\" -interaction=batchmode -halt-on-error -shell-escape"
```

**Comment by bollenn** [comment]
@albert-github I can confirm that works for the extra arguments case but does not for the environment variables case 

```
LATEX_CMD_NAME = "TEXINPUTS=$(abspath .)/latex-templates//: lualatex"
```

**Comment by albert-github** [comment]
This is indeed a bit of a difficult case as here you do a temporary environment setting  just for the `lualatex` command.
I can see that this looks like to work for the 1.9.2 version but not for the 1.9.3 version.

The only solution I can see for this, specific problem is to define a `LATEX_CMD_PRE` setting and in the Makefile we get: `$(LATEX_CMD_PRE) $(LATEX_CMD)`.
I'm not so font of a new setting but I don't see a better solution.

@bollenn do you have a (better) idea?

#doxygen what is your opinion?

**Comment by bollenn** [comment]
@albert-github  `LATEX_CMD_PRE` would very likely work but it brings extra options which might be not preferred, for sure I'm not against.

On the other hand, the spaces in path issue is a Windows only issue isn't it? is the behavior similar over there?

**Comment by albert-github** [comment]
The spaces issue is mainly a problem on Windows but due to the fact that some people think that what is possible on Windows should also be possible under Linux there will also be people using spaces in Linux paths (I myself try to stick just to either lower or upper case characters, digits and underscores, also minus signs van be a hassle).

On windows I don't know of a, real, way to use the `LATEX_CMD_PRE` other than using a (single) `&` between `%LATEX_CMD_PRE%` and `%LATEX_CMD%`. A problem would be that settings done in this way stay not just for one command but for all following commands, with possible cascading problems.

**Comment by doxygen** [comment]
> This is indeed a bit of a difficult case as here you do a temporary environment setting just for the `lualatex` command. I can see that this looks like to work for the 1.9.2 version but not for the 1.9.3 version.
> 
> The only solution I can see for this, specific problem is to define a `LATEX_CMD_PRE` setting and in the Makefile we get: `$(LATEX_CMD_PRE) $(LATEX_CMD)`. I'm not so font of a new setting but I don't see a better solution.
> 
> @bollenn do you have a (better) idea?
> 
> @doxygen what is your opinion?

I don't think more options like `LATEX_CMD_PRE` is the way to go. Before you know the config file will have to have its own shell language. If more complex things are needed why not create a wrapper batch file/shell script around the call to `lualatex` and use the LATEX_CMD to invoke the wrapper?

**Comment by albert-github** [comment]
Indeed the `LATEX_CMD_PRE` could probably occur for many more commands, and these `...PRE` would lead to a lot of extra commands and (specific?) required parsing.

The solution wit an extra wrapper script is indeed the better choice (note: one could copy this script to the latex direct by the setting `LATEX_EXTRA_FILES`).

**Comment by bollenn** [comment]
@doxygen @albert-github thx for the tip, will give the wrapper script a try later today but that might workout well...

**Comment by bollenn** [comment]
@doxygen @albert-github good alternative is to do https://github.com/doxygen/doxygen/pull/9002.

We are calling the generated Latex Makefile from our make infrastructure so we can easily set the `LATEX_CMD` from there with the correct values...

**Comment by doxygen** [comment]
@bollenn Indeed, that's a neat solution as well.

**Comment by albert-github** [comment]
I like the approach with the `?=`.
The `MKIDX_CMD` should have been done long ago!

Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by bollenn** [comment]
Thx for the feedback and merge... works smooth for me

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
